server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java

```diff
@@ -54,7 +54,7 @@
             Preconditions.checkArgument(!duration.isZero(), "'windowDuration' must be strictly positive");
 
             return flux -> flux
-                .windowTimeout(elements, duration)
```

Comment:
the use case not covered AFAIU is when you don't have a multiple of `elements`, your window never fill and the pipeline never continue to the `zipWith`. Could you write a test with less elements in the Flux than the `elements` parameter to check that case?

Reply:
I think the termination signal would do the job, as long as we don't throttle infinite streams.

Reply:
Short version: thanks for this feedback, it lead to write more tests, detect more issues and hopefully upgrade the code.